espressif/vscode-esp-idf-extension
# Feature/unit testing

## Description

Add Unit testing from ESP-IDF extension using PyTest from ESP-IDF.

TO-DO: Add fallback PyTest requirements. 

Fixes #485

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Open an example project that has esp-idf tests as described in [Unit Testing in ESP32](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/unit-tests.html). For example use the [unit test example](https://github.com/espressif/esp-idf/blob/master/tools/unit-test-app/README.md)

1. Click on "Testing" icon in the Activity Bar.
2. The tests will be automatically found in the current workspace folder.
3. Next to the test there is Run button. Pressing the button will Check if Pytest is installed, install if it is not installed, copy and build a unit test project within the current project, execute PyTest and parse the results.

- Expected behaviour: Tests output is shown as Task Output and the result is shown in the Testing UI (passed/ failed)

- Expected output: Test execution output is shown as task output.

## How has this been tested?

- Manual Testing with example unit test app project.

**Test Configuration**:
* ESP-IDF Version: 5.0
* OS (Windows,Linux and macOS): macOS

## Dependent components impacted by this PR:

- PyTest installed from ESP-IDF

## Checklist
